### ISSUE TITLE
parsl workqueue setup dependency does not work

### ISSUE THREAD

**Issue by benclifford** [Parsl/parsl#1158]
`pip install .[workqueue]` fails, at least in CI, even if work queue is already installed by:

```
    - bash parsl/executors/workqueue/install-workqueue.sh
    - export PATH=$PATH:/tmp/cctools/bin
    - export PYTHONPATH=/tmp/cctools/lib/python3.5/site-packages
```

PR #1155 attempts to rearrange dependencies, and I encountered this as part of testing that PR.

Should the `workqueue` `setup.py` dependency go away? Will it work eventually?

cc @dthain (I can't assign you this ticket in the github UI)

**Comment by dthain** [comment]
@tjdasso can you help out here?   (AFAIK there is no pip  workqueue module just yet.)

**Comment by tjdasso** [comment]
As of right now, there is no pip module for workqueue. @eamonlm is currently working on this at the moment. If I remember correctly, he is making the entire CCTools collection of software installable via pip, i.e., `pip install cctools`, instead of `pip install workqueue`. 

I just tested the install-workqueue.sh script locally, and it still seems to work, assuming $CCTOOLS_VERSION is set to an appropriate version (I set it to "7.0.16"). I think Eamon is relatively close to getting the pip module to work, so cctools should be able to be added as a dependency somewhat soon.

**Comment by benclifford** [comment]
OK. So maybe for now instructions are run `install-workqueue.sh`, and `setup.py` should have the (currently broken) workqueue dependency block removed; then when the stack is installable via pip, swap the install-workqueue stuff back out for pip-based dependency management?

**Comment by tjdasso** [comment]
Yes, that sounds good. FWIW, Eamon finished getting a cctools installation set up with conda, so `conda install -c cclnd cctools` should also be able to install the full software. I don't know if that might be another viable option for now.

**Comment by dthain** [comment]
Returning to an old thread...

`conda install -c conda-forge ndcctools` is the current recommended way to get the cctools software, we are no longer doing `pip`.

The current `install-workqueue.sh` script is downloading the binary cctools and running that directly.

**Comment by dthain** [comment]
@yadudoc could you comment on what the desired approach for installation during testing is?

**Comment by benclifford** [comment]
The CI installs stuff into the main user environment. I think I'd prefer to install workqueue specific stuff into that environment in as close a way as possible to what a user would be told to do.
